### PR TITLE
Adding isAppInstalled logic

### LINKED ISSUE #1526: [IOU] Add isAppInstalled logic to Expensify.cash
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/master/CONTRIBUTING.md) for onboarding!
___

view the post on Upwork [here](https://www.upwork.com/jobs/~0152f89613f3e316fe).

**Platform - version:**  
current

We require a method of detecting installed apps for the Expensify.cash app on Android & iOS.

1. Install and implement the [react-native-check-app-install](https://github.com/redpandatronicsuk/react-native-check-app-install) package. 
2. Create a helper function under libs `isAppInstalled()` (similar to [getPlatform](https://github.com/Expensify/Expensify.cash/tree/master/src/libs/getPlatform)) which takes two params: `iosScheme, androidPackageId`, and returns true if the selected app is installed.
3. Within the helper function, `getPlatform()` should be used to determine which library function to use: 
    -  **Android:** `AppInstalledChecker.isAppInstalledAndroid(androidPackageId)`
    - **iOS:** `AppInstalledChecker.checkURLScheme(iosScheme)`
4. As this is only used on mobile, this should be build within an `index.native.js` file (unfortunately the params are different for iOS/Android, so we cant create `index.android.js|`index.ios.js``). Also, the default `index.js` function should always return false.

### PR DESCRIPTION
**Pull request by SameeraMadushan**
### Details
This PR includes a method of detecting installed apps for the Expensify.cash app on Android & iOS.

#### iOS only
As discussed in the issue, currently added `venmo` app `URLscheme` on iOS.
If more apps need to be added, Then I can add them to `LSApplicationQueriesSchemes`  under `info.plist`

#### Android
Does not need additional steps to be done with new apps.

### Fixed Issues
Fixes: https://github.com/Expensify/Expensify.cash/issues/1526

#### iOS and Android

- After importing `isAppInstalled` function, it will execute from `index.native.js`
- The function returns a promise with a boolean result (`true` or `false`)
- To call the function use `async` `await`  ( eg: `await isAppInstalled('venmo')`)

#### Desktop and Web

- The function will execute from `index.js`
- It will always return `false`

### Tested On

- [x] Web
- [x] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### CODE REVIEW

**Comment by SameeraMadushan** [comment]
I have read the CLA Document and I hereby sign the CLA

File: src/libs/isAppInstalled/index.native.js [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+/**
+ * Gets the URLSchema and check the existence of that application
+ *
+ * @param {String} urlScheme
+ * @returns {Promise<Boolean>}
+ */
+export default async function isAppInstalled(urlScheme) {
+    try {
+        return await Linking.canOpenURL(`${urlScheme}://`);
```
Comment (Julesssss) [inline src/libs/isAppInstalled/index.native.js:11]: Could we replace await with Promise (for consistency throughout the codebase). See [this STYLE.md section](https://github.com/Expensify/Expensify.cash/blob/460f9a2040d1a4c4a14cb308557a9a50b50a0123/STYLE.md#esnext-are-we-allowed-to-use-insert-new-language-feature-why-or-why-not) for more info.

Comment (SameeraMadushan) [reply src/libs/isAppInstalled/index.native.js]: Hi @Julesssss 
Added commit with converting async-await to a promise.
When calling the function developer can use then().catch() or async-await

**Review by Luke9389 — APPROVED** [review]

**Comment by Julesssss** [comment]
Verified that the function works as expected on iOS, Android & Web 👍

**Review by Julesssss — APPROVED** [review]

**Pull request merged**